### PR TITLE
add: iminuit

### PR DESCRIPTION
**Pull request by henryiii**
### Description

I started this a long time ago, and thought I'd see if it works now that iminuit is using scikit-build-core.

### Checklists

- Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- Add new / update outdated documentation

### CODE REVIEW

File: packages/iminuit/test_iminuit.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import pytest
+from pytest_pyodide import run_in_pyodide
+
+
+@run_in_pyodide(packages=["iminuit"])
+def test_iminuit(selenium):
+    import iminuit
```
Comment (hoodmane) [inline packages/iminuit/test_iminuit.py:7]:
```suggestion
    import pytest
    import iminuit
```

File: packages/iminuit/test_iminuit.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    import iminuit.cost
+    import numpy as np
```
Comment (hoodmane) [inline packages/iminuit/test_iminuit.py:9]:
```suggestion
    import iminuit
    import iminuit.cost
    import numpy as np
    import pytest
```

File: packages/iminuit/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package:
+  name: iminuit
+  version: 2.25.2
+source:
+  url: https://files.pythonhosted.org/packages/78/f1/416e559122a3872878853429a87143b4b4f5d33e23c1317be8a270e875aa/iminuit-2.25.2.tar.gz
+  sha256: "3bf8a1b96865a60cedf29135f4feae09fa7c66237d29f68ded64e97a823a9b3e"
```
Comment (agriyakhetarpal) [inline packages/iminuit/meta.yaml:6]:
```suggestion
package:
  name: iminuit
  version: 2.28.0
source:
  url: https://files.pythonhosted.org/packages/d8/c3/9385361c5f115928c1490422d9d824581141f8f444096a6ed00009add74e/iminuit-2.28.0.tar.gz
  sha256: "6646ae0b66a4760e02cd73711d460a6cf2375382b78ce8344141751595596aad"
```

**Review by agriyakhetarpal — COMMENTED** [review]
Maybe this works now with `iminuit==2.28.0` out? I'll update the branch here

**Comment by hoodmane** [comment]
I think Python does support non-ascii characters as long as they are unicode letters. It's just a lint telling you not to use them in identifiers.

**Comment by henryiii** [comment]
Yes, that's correct. Annoying linter rule, since these are quite valid. In this case, it's a test to make sure it reads unicode args correctly.

**Comment by agriyakhetarpal** [comment]
Weird – I don't see a build error in there, but it looks more related to some missing config?

**Comment by agriyakhetarpal** [comment]
I guess a 64-bit generator needs to be passed, but I'm not entirely sure how.

**Comment by agriyakhetarpal** [comment]
and I found one of your older discussion threads: https://github.com/pyodide/pyodide/discussions/2186 😃

**Comment by henryiii** [comment]
Sure, you can do it. We will probably start adding a warning message or something like that from pybind11 if it’s not set.

**Comment by agriyakhetarpal** [comment]
That sounds great. I shall do this later in the day today, then!

**Comment by agriyakhetarpal** [comment]
Upstream PR: https://github.com/scikit-hep/iminuit/pull/1023 and patch: 4f45d9e59ea7d692bf3a41a440aad31351917d91

**Comment by agriyakhetarpal** [comment]
[`iminuit` is building successfully](https://app.circleci.com/pipelines/github/pyodide/pyodide/7716/workflows/6085fcd7-1b6a-4924-bba5-17bee90fd6ec/jobs/110839?invite=true#step-104-15031_80), but [`test_iminuit.py` is failing](https://app.circleci.com/pipelines/github/pyodide/pyodide/7716/workflows/6085fcd7-1b6a-4924-bba5-17bee90fd6ec/jobs/110850?invite=true#step-102-161176_52) – it needs `packaging` as a dependency

**Review by agriyakhetarpal — APPROVED** [review]
The test passes for [Node.js](https://app.circleci.com/pipelines/github/pyodide/pyodide/7719/workflows/973e341b-5e8f-43e2-897c-4d2fc5963ce9/jobs/110935?invite=true#step-102-107665_79) and [Chrome](https://app.circleci.com/pipelines/github/pyodide/pyodide/7719/workflows/973e341b-5e8f-43e2-897c-4d2fc5963ce9/jobs/110933?invite=true#step-102-107979_79). Thank you, @henryiii!

**Pull request merged**